logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/TypingIndicatorIncomingRepository.kt

```diff
@@ -55,36 +54,27 @@ internal class TypingIndicatorRepositoryImpl(
 
     override suspend fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId) {
         userTypingCache.block { entry ->
-            entry[conversationId]?.apply { this.removeAll { it.userId == userId } }
+            entry[conversationId]?.apply { this.removeAll { it == userId } }
         }.also {
             userTypingDataSourceFlow.tryEmit(Unit)
         }
     }
 
-    override suspend fun observeUsersTyping(conversationId: ConversationId): Flow<Set<ExpiringUserTyping>> {
+    override suspend fun observeUsersTyping(conversationId: ConversationId): Flow<Set<UserId>> {
         return userTypingDataSourceFlow
             .map { userTypingCache[conversationId] ?: emptySet() }
             .onStart { emit(userTypingCache[conversationId] ?: emptySet()) }
     }
 
-    companion object {
-        const val BUFFER_SIZE = 32 // drop after this threshold
-    }
-}
-
-// todo expire by worker
-data class ExpiringUserTyping(
-    val userId: UserId,
-    val date: Instant
-) {
-    override fun equals(other: Any?): Boolean {
-        return other != null && when (other) {
-            is ExpiringUserTyping -> other.userId == this.userId
-            else -> false
+    override suspend fun clearExpiredTypingIndicators() {
+        userTypingCache.block { entry ->
+            entry.clear()
+        }.also {
+            userTypingDataSourceFlow.tryEmit(Unit)
```

Comment:
This is so we can make the Observers of `userTypingDataSourceFlow`  `collect` the new state. Since we are relying on this flow to get new map `userTypingCache` states.